### DISCUSSION TITLE
How to accurately scan the version of dependencies?

### DISCUSSION THREAD

**Discussion by zangcc** [aquasecurity/trivy#4864] [Q&A]
### Question

How to accurately scan the version of dependencies?

Because the version I scanned with trivy is different from the actual version，The following is the result I scanned with the trivy command：
`./trivy rootfs --severity CRITICAL,HIGH /root/code/codeproject -o rootfs.txt`
![image](https://github.com/aquasecurity/trivy/assets/64825932/1272bed8-3428-4727-85e1-bbfdc8ea5ae0)

Here are the actual dependency versions of my code project:
![image](https://github.com/aquasecurity/trivy/assets/64825932/51347211-68c9-44eb-92ff-63977c06e162)

Does trivy have any parameters to accurately scan the dependent version?

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Version

```bash
Version: 0.43.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-07-20 12:11:37.696263932 +0000 UTC
  NextUpdate: 2023-07-20 18:11:37.696263532 +0000 UTC
  DownloadedAt: 2023-07-24 07:49:06.099887061 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-07-24 00:52:18.487469647 +0000 UTC
  NextUpdate: 2023-07-27 00:52:18.487469147 +0000 UTC
  DownloadedAt: 2023-07-24 07:52:17.429868261 +0000 UTC
```

**Comment by ba1ma0** [comment]
I also met the same problem ! Pls....Any solution for this problem? @umax @shift @simon-engledew @ocean @zangcc

**Reply by DmitriyLewen** [reply]
Hello @zangcc, @ba1ma0 

Looks like your `safeherom-gm-mpc-service-gateway-web-1.0-SNAPSHOT.jar` file contains `org.springframework:spring-core v5.3.6`.
Use `-f json` to see full path to dependency (information [here](https://aquasecurity.github.io/trivy/v0.43/docs/scanner/vulnerability/language/java/#jarwarparear)).

**Comment by AnaisUrlichs** [comment]
@DmitriyLewen could you please take a look?
